docs/development.md

```diff
@@ -382,8 +439,30 @@ The `lang` property of the `<script>` block is used to specify the used programm
  * for Javascript, use `<script lang="javascript"> ... </script>`
  * for Python, use `<script lang="python"> ... </script>`
 
-`<script>` also supports `tags`. For more information, see the dedicated section **Plugins and tags**.
+`<script>` also supports `tags`. For more information, see the dedicated section for [`tags`](development?id=tags).
+
+## Use custom libraries
+You can specify plugin requirements of ImJoy in the dedicated field in
+its `config` [block](http://localhost:8000/docs#/development?id=requirements)
+
+This allows you include public libraries with `pip` or `conda`, but also your own
+libraries. Below we describe some of the most commonly encountered scenarios.
 
+1.  Your python module is deployed to as a pip repository (`pip.pypa.io`). You can
+    then add its pip name to `requirements`.
+0.  You host the source code and `setup.py` on GitHub. Several excellent ressources,
+    such as [this one](https://packaging.python.org/tutorials/packaging-projects/), explain
+    how to specify this file. You can add  `"pip:git+https://github.com/myUserName/myRepo#egg=myRepo"` to `requirements`.
+    Note that the `#egg=myRepo` allows to determine the [GitHub release](https://help.github.com/articles/creating-releases/)
+    of your library, controlling which version is installed.
+0.  You host your source code and a `requirements.txt` on GitHub.
+    Add to `requirements`: `repo: https://github.com/XXXX` and `cmd: pip install -r XXXXX/requirements.txt`
+0.  You host the source code and a `environment.yml` defining the virtual env with conda and pip
+    dependencies. You can add the repo with `repo: https://github.com/XXXX` and install the
+    environment with `"env": "conda env create -f XXXX/environment.yml"` imjoy cannot use `environment.yml` directly, need to convert to imjoy `requirements`.
```

Comment:
This is already supported. So the user can just provide that file, and add `"env": "conda env create -f XXXX/environment.yml"` to the requirements.

Reply:
essentially we need to delete the second part of the sentence: " imjoy cannot use `environment.yml` directly, need to convert to imjoy `requirements`"